### PR TITLE
feat: The 'Save cart for later' button in <cx-add-to-saved-cart> is disabled if its current cart has no entries

### PR DESCRIPTION
**Pull request by justinlee01**
#Closes https://jira.tools.sap/browse/CXSPA-527

### CODE REVIEW

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html [hunk 1]
```diff
@@ -12,6 +12,7 @@
       <span>{{ 'addToSavedCart.savedCarts' | cxTranslate }}</span>
     </button>
     <button
+      *ngIf="showSaveCartForLater(cart)"
```
Comment (Zeyber) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html]: It would be better if we use an observable here. Otherwise the function will keep being recalled.

**Review by Zeyber — CHANGES_REQUESTED** [review]

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
       <span>{{ 'addToSavedCart.savedCarts' | cxTranslate }}</span>
     </button>
     <button
+      *ngIf="showSaveCartForLater$ | async"
```
Comment (justinlee01) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html]: Does this need to be feature flagged for 5.1? It's a simple fix, just was wondering.

Comment (Zeyber) [reply feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html]: Perhaps we can disable this button instead of removing it from the template. It is not necessarily a breaking change in this case.

**Review by Zeyber — CHANGES_REQUESTED** [review]

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html (hunk 2 above)
Comment (justinlee01) [reply feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html]: My only concern is that the customer may think "saving the cart for later" is a possible thing to do with an empty cart, however they have not fulfilled some invisible condition to do so.
Nevertheless I will update the PR with the proposed change. Perhaps my concern is minor.

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.spec.ts [hunk 3]
```diff
@@ -106,6 +108,75 @@ describe('AddToSavedCartComponent', () => {
     );
   });
 
+  it("should show the 'Save cart for later' button", (done) => {
```
Comment (Zeyber) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.spec.ts]: Semantaics, but this should be `enable/disable` now instead of `show/hide`.

**Review by Zeyber — CHANGES_REQUESTED** [review]
I think it could be a good call to add/modify e2e tests to make sure the user cannot save the cart in the case of "empty cart" as this is a UI improvement.

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.spec.ts (hunk 3 above)
Comment (justinlee01) [reply feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.spec.ts]: Ah, slipped my mind. Thank you.

**Comment by justinlee01** [comment]
@Zeyber I have spent the last few days trying to add an E2E test for this feature. I will explain the abnormal difficulty.

I have implementing this test plan:
1. Add a product to cart.
2. Go to cart. Check that "Save cart for later" is enabled.
3. Save the product for later. Check that "Save cart for later" is disabled.

When executing step 3, the code is trying to update the saved cart, which does not exist yet. Because of NgRX state management, this creates a malformed HTTP request causing the program to terminate early. I have not been able to reproduce it outside of Cypress.

I can continue pursuing this if it is absolutely necessary. A possible solution is to simply not allow the program to query for undefined saved carts.

**Comment by justinlee01** [comment]
Finally found the issue. When the application loads, it also loads the selective (saved) cart.

Because Cypress is so quick, it creates a race hazard where UI is loaded and interacted with, but the saved cart has not been loaded yet, thus causing issues. The solution is to use Angular's router to navigate through routes instead of using `cy.visit` which triggers the application's loading code again.

**Review by Zeyber — CHANGES_REQUESTED** [review]
I have found this test to fail on my local environment. Should something be done for setup?
image

Also, this should really be a part of `save-for-later` e2es rather than the main `cart` e2es as it is an additional feature.

**Comment by justinlee01** [comment]
@Zeyber I'm not sure what's missing from the set-up; if you can share the full URL of the request that is responded with 400, that may help. That request + error looks familiar.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ describe('Cart', () => {
     });
 
     context('Registered user', () => {
-      before(() => {
+      beforeEach(() => {
```
Comment (Zeyber) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts]: Leave as is?

**Review by Zeyber — CHANGES_REQUESTED** [review]

**Review by Zeyber — APPROVED** [review]

**Review by kimhw0630 — APPROVED** [review]
